Packs/Cisco-umbrella/Integrations/CiscoUmbrellaEnforcement/CiscoUmbrellaEnforcement.py

```diff
@@ -0,0 +1,188 @@
+        if request:
+            res = self._http_request('GET', full_url=request)
+        else:
+            res = self._http_request('GET', url_suffix=f"domains?customerKey={self.api_key}&{suffix}")
+        return res
+
+    def delete_domains(self, domain_name: str, domain_id: str):
+        res = ''
+        if domain_id:
+            res = self._http_request('DELETE', f"domains/{domain_id}?customerKey={self.api_key}",
+                                     return_empty_response=True)
+        if domain_name:
+            res = self._http_request('DELETE', f"domains?customerKey={self.api_key}&where[name]={domain_name}",
+                                     return_empty_response=True)
+        return res
+
+    def add_event_to_domain(self, event: dict):
+        return self._http_request('POST', f"events?customerKey={self.api_key}", json_data=event,
+                                  return_empty_response=True)
+
+
+def domains_list_suffix(page: Optional[str] = '', limit: Optional[str] = '', request: Optional[str] = '') -> str:
+    """
+    Create the relevant suffix for the domains command,
+     Either there is a complete request that should be sent or page and limit arguments.
+    :param page: The number of the requested page for domains command.
+    :param limit: The limit of the queries to return from the domains command.
+    :param request: Full request to send to API, Used when running the domains list command and there is a large number
+                    of queries which are then returned with paging and have a next page request built already.
+    :return: (str) with the suffix.
+    """
+    suffix = ''
+    if request:
+        suffix = request
+    if page:
+        suffix += f'page={page}&'
+    if limit:
+        suffix += f'limit={limit}&'
+    return suffix
+
+
+def create_domain_list(client: Client, response: dict) -> list:
+    """
+    :param client: Cisco Umbrella Client for the api request.
+    :param response: the first response of the domains api call.
+    :return: list of all of the domains from the api.
+    """
+    full_domains_list = []
+    while response:
+        response_data = response.get('data', [])
+        for domain in response_data:
+            full_domains_list.append(domain.get('name'))
+        response_next_page = response.get('meta', {}).get('next', '')
+        response = client.get_domains_list(response_next_page) if response_next_page else {}
+    return full_domains_list
+
+
+def domains_list_command(client: Client, args: dict) -> CommandResults:
+    """
+    :param client: Cisco Umbrella Client for the api request.
+    :param args: args from the user for the command.
+    """
+    page = args.get('page', '')
+    limit = args.get('limit', '')
+    suffix = domains_list_suffix(page=page, limit=limit)
+    response = client.get_domains_list(suffix=suffix)
+    domains_list = create_domain_list(client, response)
```

Comment:
not sure i follow why you call `client.get_domains_list` from here and from `create_domain_list`

Reply:
The reason for that is because when there are many domains they are returned in pages and after a full page there is a next param that says what is the next page ,callling the next page is done using get_domains_list.